### PR TITLE
Add native support for iLLeniumStudios's fivem-appearance

### PR DESCRIPTION
**Pull request by Z3rio**
**Describe Pull request**
This pull request adds native support for ILLeniumStudio's clothing script (fivem-appearance) which is quite nice to have since alot of people use it instead of qb-clothing.
 
If your PR is to fix an issue mention that issue here
#155

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes/no] (Be honest)
I have tested it with qb-clothing and partially with fivem-appearance
- Does your code fit the style guidelines? [yes/no]
Yeah
- Does your PR fit the contribution guidelines? [yes/no]
Yep

### CODE REVIEW

**Review by Waves966 — COMMENTED** [review]
This isn't really needed as his script has backwards compatibility with qb-clothing and works fine with multicharacter.

**Comment by Z3rio** [comment]
> This isn't really needed as his script has backwards compatibility with qb-clothing and works fine with multicharacter.

Nope, it lacks the events and such to spawn the character shown whilst clicking a character.
I wouldn't have added it if it wasn't needed. Search it up yourself if you want to ;)

**Comment by Waves966** [comment]
This is all you need to do to replicate the same effect of what you're doing.
```lua
RegisterNUICallback('cDataPed', function(nData, cb)
    local cData = nData.cData
    SetEntityAsMissionEntity(charPed, true, true)
    DeleteEntity(charPed)
    if cData ~= nil then
        QBCore.Functions.TriggerCallback('qb-multicharacter:server:getSkin', function(model, data)
            model = model ~= nil and tonumber(model) or false
            if model ~= nil then
                CreateThread(function()
                    RequestModel(model)
                    while not HasModelLoaded(model) do
                        Wait(0)
                    end
                    charPed = CreatePed(2, model, Config.PedCoords.x, Config.PedCoords.y, Config.PedCoords.z - 0.98, Config.PedCoords.w, false, true)
                    SetPedComponentVariation(charPed, 0, 0, 0, 2)
                    FreezeEntityPosition(charPed, false)
                    SetEntityInvincible(charPed, true)
                    PlaceObjectOnGroundProperly(charPed)
                    SetBlockingOfNonTemporaryEvents(charPed, true)
                    data = json.decode(data)

                    if GetResourceState('fivem-appearance') == 'started' then
                        exports['fivem-appearance']:setPedAppearance(charPed, data)
                    else
                        TriggerEvent('qb-clothing:client:loadPlayerClothing', data, charPed)
                    end
                end)
            else
                CreateThread(function()
                    local randommodels = {
                        "mp_m_freemode_01",
                        "mp_f_freemode_01",
                    }
                    model = joaat(randommodels[math.random(1, #randommodels)])
                    RequestModel(model)
                    while not HasModelLoaded(model) do
                        Wait(0)
                    end
                    charPed = CreatePed(2, model, Config.PedCoords.x, Config.PedCoords.y, Config.PedCoords.z - 0.98, Config.PedCoords.w, false, true)
                    SetPedComponentVariation(charPed, 0, 0, 0, 2)
                    FreezeEntityPosition(charPed, false)
                    SetEntityInvincible(charPed, true)
                    PlaceObjectOnGroundProperly(charPed)
                    SetBlockingOfNonTemporaryEvents(charPed, true)
                end)
            end
            cb("ok")
        end, cData.citizenid)
    else
        CreateThread(function()
            local randommodels = {
                "mp_m_freemode_01",
                "mp_f_freemode_01",
            }
            local model = joaat(randommodels[math.random(1, #randommodels)])
            RequestModel(model)
            while not HasModelLoaded(model) do
                Wait(0)
            end
            charPed = CreatePed(2, model, Config.PedCoords.x, Config.PedCoords.y, Config.PedCoords.z - 0.98, Config.PedCoords.w, false, true)
            SetPedComponentVariation(charPed, 0, 0, 0, 2)
            FreezeEntityPosition(charPed, false)
            SetEntityInvincible(charPed, true)
            PlaceObjectOnGroundProperly(charPed)
            SetBlockingOfNonTemporaryEvents(charPed, true)
        end)
        cb("ok")
    end
end)
```

**Comment by Bama94** [comment]
Or could you not make a config.clothingoption where config.clothing = 'fivem-appearance' or 'qb-clothing'

**Comment by Z3rio** [comment]
> Or could you not make a config.clothingoption where config.clothing = 'fivem-appearance' or 'qb-clothing'

Why do that when it can be done automatically through this instead?

File: client/main.lua [hunk 1]
```diff
@@ -2,6 +2,11 @@ local cam = nil
 local charPed = nil
 local QBCore = exports['qb-core']:GetCoreObject()
 
+local clothingScript = "qb-clothing"
+if GetResourceState("fivem-appearance") == "started" then
```
Comment (Irishstevie) [inline client/main.lua:6]: This is no longer called fivem-appearance

Comment (nzkfc) [reply client/main.lua:6]: Actually fivem-appearance is the old resource but people still use it, illenium-appearance is just the new one.

Comment (Z3rio) [reply client/main.lua:6]: Yeah, I'm unsure if support for illenium-appearance is even the same as support for fivem-appearance.
So changing it from fivem-appearance to illenium-appearance without looking over the rest of the code would perhaps be an bad idea.

**Pull request closed without merging**